modules/product/src/main/java/com/opengamma/strata/product/fx/ResolvedFxVanillaOption.java

```diff
@@ -123,12 +90,31 @@ public LocalDate getExpiryDate() {
   }
 
   /**
-   * Gets the currency on which the payoff occurs. 
+   * Gets the strike rate. 
+   * 
+   * @return the strike
+   */
+  public double getStrike() {
+    return Math.abs(underlying.getCounterCurrencyPayment().getAmount() /
+        underlying.getBaseCurrencyPayment().getAmount());
+  }
+
+  /**
+   * Gets the put/call
```

Comment:
Returns the put/call flag. This is the put/call for the base currency. If the amount for the base currency is positive, the option is a call on the base currency (put on counter currency) and if the amount for the base currency is negative, the option is a put on the base currency (call on counter currency).